Signal/src/contact/OWSContactsManager.m

```diff
@@ -193,17 +210,16 @@ - (void)updateWithContacts:(NSArray<Contact *> *)contacts
 
             [self.avatarCache removeAllImages];
 
-            [self intersectContacts];
-
-            [self buildSignalAccounts];
+            [self intersectContactsWithCompletion:^(NSError *_Nullable error) {
+                [self buildSignalAccountsAndClearStaleCache:clearStaleCache];
+            }];
         });
     });
 }
 
-- (void)buildSignalAccounts
+- (void)buildSignalAccountsAndClearStaleCache:(BOOL)clearStaleCache;
 {
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSMutableDictionary<NSString *, SignalAccount *> *signalAccountMap = [NSMutableDictionary new];
+    dispatch_async(self.serialQueue, ^{
```

Comment:
Doing this concurrently could give us strange results since it uses multiple transactions (there's good reason for using multiple transactions, most of them are reads, and only the writes intend to be narrowly scoped)